projects/js-packages/components/components/threats-data-views/index.tsx

```diff
@@ -475,14 +476,37 @@ export default function ThreatsDataViews( {
 		isThreatEligibleForUnignore,
 	] );
 
+	const sortedData = useMemo( () => {
+		// If no statuses are set, return the data as is
+		if ( data.every( item => item.status === null ) ) {
+			return data;
+		}
+
+		const statusOrder = [ 'current', 'fixed', 'ignored' ];
+
+		return [ ...data ].sort( ( a, b ) => {
+			// Compare by status
+			const statusComparison = statusOrder.indexOf( a.status ) - statusOrder.indexOf( b.status );
+			if ( statusComparison !== 0 ) {
+				return statusComparison;
+			}
+
+			// If statuses are the same, compare by severity
+			const severityA = a.severity ?? -1;
+			const severityB = b.severity ?? -1;
+
+			return severityB - severityA;
+		} );
+	}, [ data ] );
```

Comment:
Could this be accomplished by adjusting `Scan_Status::sort_threats` on [the server side](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/protect-status/src/class-scan-status.php#L315)?

Reply:
>Could this be accomplished by adjusting Scan_Status::sort_threats on [the server side](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/protect-status/src/class-scan-status.php#L315)?

The list will sometimes only be historical threats or additional include historical threats, we'd need a way to combine them before sorting as we only do so currently on the client side before passing the set to the component.